File: daily_challenge_leetcode/3100.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var maxBottlesDrunk = function (numBottles, numExchange) {
  let empty = numBottles;
  let drunk = numBottles;
  while (empty >= numExchange) {
    empty -= numExchange;
    empty++;
    drunk++;
    numExchange++;
  }
  return drunk;
};


//  Sridhara Acharya’s / quadratic formula
var maxBottlesDrunk = function (numBottles, numExchange) {
  return Math.floor(
    numBottles +
      (-2 * numExchange +
        3 +
        Math.sqrt(
          4 * numExchange * numExchange + 8 * numBottles - 12 * numExchange + 1
        )) /
        2
  );
};

File: daily_challenge_leetcode/3100.py
class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        bottleDrunk = numBottles
        emptyBottles = numBottles

        while emptyBottles >= numExchange:
            emptyBottles -= numExchange
            numExchange += 1
            bottleDrunk += 1
            emptyBottles += 1
        return bottleDrunk


import math


# Sridhara Acharya’s / quadratic formula
class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        return int(
            numBottles
            + (
                (
                    -2 * numExchange
                    + 3
                    + math.sqrt(
                        4 * numExchange * numExchange
                        + 8 * numBottles
                        - 12 * numExchange
                        + 1
                    )
                )
                / 2
            )
        )
